woocommerce/woocommerce
# CYS - AI flow: Replace recordEvent with trackEvent

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR updates how we track the events for the CYS: when the user is through the `entrepreneur flow`, it is necessary to `ref:entrepreneur-signup` as an additional parameter.

Closes #46838 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure that you are on a WooCommerce site where the CYS AI flow works.
2. Ensure that you're running a production build.
3. Open the network tools.
4. Visit /wp-admin/admin.php?page=wc-admin&path=%2Fcustomize-store%2Fdesign-with-ai&ref=entrepreneur-signup.
5. Search `t.gif` in the network tools filter -> ![image](https://github.com/woocommerce/woocommerce/assets/4463174/e2d9d714-0d38-4ef9-9721-4abdcb3baaa7)
6. Ensure that events are tracked and the `ref:entrepreneur-signup` is present.
7. Visit /wp-admin/admin.php?page=wc-admin&path=%2Fcustomize-store%2Fdesign-with-ai.
8. Search `t.gif` in the network tools filter -> ![image](https://github.com/woocommerce/woocommerce/assets/4463174/e2d9d714-0d38-4ef9-9721-4abdcb3baaa7)
9. Ensure that events are tracked and the `ref:entrepreneur-signup` is not present.


### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

CYS - AI Flow: replace recordEvent with trackEvent.

</details>
